config/config.go

```diff
@@ -0,0 +1,140 @@
+package config
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/jeremywohl/flatten"
+	"github.com/mcuadros/go-defaults"
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/viper"
+)
+
+type Loader struct {
+	v *viper.Viper
+}
+
+type LoaderOption func(*Loader)
+
+func WithViper(in *viper.Viper) LoaderOption {
+	return func(l *Loader) {
+		l.v = in
+	}
+}
+
+func WithConfigName(in string) LoaderOption {
+	return func(l *Loader) {
+		l.v.SetConfigName(in)
+	}
+}
+
+func AddConfigPath(in string) LoaderOption {
+	return func(l *Loader) {
+		l.v.AddConfigPath(in)
+	}
+}
+
+func WithConfigType(in string) LoaderOption {
+	return func(l *Loader) {
+		l.v.SetConfigType(in)
+	}
+}
+
+func WithEnvKeyReplacer(old string, new string) LoaderOption {
+	return func(l *Loader) {
+		l.v.SetEnvKeyReplacer(strings.NewReplacer(old, new))
+	}
+}
+
+func NewLoader(options ...LoaderOption) *Loader {
+	loader := &Loader{
+		v: getViperWithDefaults(),
+	}
+
+	for _, option := range options {
+		option(loader)
+	}
+	return loader
+}
+
+// Load loads configuration into the given mapstructure (https://github.com/mitchellh/mapstructure)
+// from a config.yaml file and overrides with any values set in env variables
+func (l *Loader) Load(config interface{}) {
```

Comment:
Any reason we choose to `panic` instead of returning an `error` across this method?

Reply:
The plan was that if the configuration is not loaded the application can not be started, but will leave that for the application to decide. Will return the error from here for application to handle.